### PR TITLE
[Merged by Bors] - drop proposals log and add metrics

### LINKED ISSUE #4501: proposal handler: reduce log spam
## Description
especially with fetch error below. replace log with metrics of errors

Screenshot from 2023-06-10 18-18-25

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
Closes #4501 

## Changes
- debug log all handling error
- log error on unexpected database error
- add metrics per error

### CODE REVIEW

File: proposals/metrics.go [hunk 1]
```diff
@@ -82,3 +82,38 @@ var (
...
+	failedInit     = processErrors.WithLabelValues(eInit)
+	known          = processErrors.WithLabelValues(eKnown)
+	preGenesis     = processErrors.WithLabelValues(eTooOld)
+	badSigProposal = processErrors.WithLabelValues(eBadSigP)
+	badSigBallot   = processErrors.WithLabelValues(eBadSigB)
+	badData        = processErrors.WithLabelValues(eBadData)
+	unavailRef     = processErrors.WithLabelValues(eAvail)
+	badVote        = processErrors.WithLabelValues(eVote)
+	notEligible    = processErrors.WithLabelValues(eElig)
+	failedPublish  = processErrors.WithLabelValues(ePublish)
```
Comment (dshulyak) [inline proposals/metrics.go]: there is really no need for those constants. as you are using vars, so kinds can be defined in place

File: proposals/metrics.go [hunk 2]
```diff
@@ -82,3 +82,38 @@ var (
...
+
+var (
+	processErrors = metrics.NewCounter(
+		"errs",
+		subsystem,
+		"number of errors",
+		[]string{"kind"},
+	)
+	malformed      = processErrors.WithLabelValues(eMalformed)
+	failedInit     = processErrors.WithLabelValues(eInit)
```
Comment (dshulyak) [inline proposals/metrics.go]: "init" ?

int seems confusing

File: proposals/metrics.go [hunk 3]
```diff
@@ -82,3 +82,38 @@ var (
...
+	known          = processErrors.WithLabelValues(eKnown)
+	preGenesis     = processErrors.WithLabelValues(eTooOld)
```
Comment (dshulyak) [inline proposals/metrics.go]: "genesis" ? tooOld seems off

**Review by dshulyak — APPROVED** [review]

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**